crawler/plugins/applications/apache/apache_crawler.py

```diff
@@ -0,0 +1,80 @@
+
+
+import urllib2
+from plugins.applications.apache import feature
+from collections import defaultdict
+from utils.crawler_exceptions import CrawlError
+
+
+def retrieve_status_page(host, port):
+    statusPage = "http://%s:%s/server-status?auto" % (host, port)
+    req = urllib2.Request(statusPage)
+    response = urllib2.urlopen(req)
+    return response.read()
+
+
+def parse_score_board(line, stats):
+    switch = {
+        "_": 'waiting_for_connection',
+        "S": 'starting_up',
+        "R": 'reading_request',
+        "W": 'sending_reply',
+        "K": 'keepalive_read',
+        "D": 'dns_lookup',
+        "C": 'closing_connection',
+        "L": 'logging',
+        "G": 'graceful_finishing',
+        "I": 'idle_worker_cleanup',
+    }
+    res = line.split(': ')
+
+    workcounts = defaultdict(int)
+    for i in res[1]:
+        workcounts[i] += 1
+
+    for x, y in workcounts.iteritems():
+        stats[switch.get(x)] = str(y)
+
+
+def retrieve_metrics(host='localhost', port=80):
+    try:
+        status = retrieve_status_page(host, port).splitlines()
+    except Exception:
+        raise CrawlError("can't access to http://%s:%s",
+                         host, port)
+    switch = {
+        "Total kBytes": 'Total_kBytes',
+        "Total Accesses": 'Total_Accesses',
+        "BusyWorkers": "BusyWorkers",
+        "IdleWorkers": "IdleWorkers",
+        "BytesPerSec": "BytesPerSec",
+        "BytesPerReq": "BytesPerReq",
+        "ReqPerSec": "ReqPerSec",
+        "Uptime": "Uptime"
+    }
+
+    stats = {}
+
+    line_num = 0
+    for line in status:
+        line_num += 1
```

Comment:
line_num is not used anywhere. Thank you for useful comments.I will remove this variable when I create a new PR to add a new application plugin to monitor tomcat.